### PR TITLE
feat(Templates): Update aws-nodejs-typescript template

### PR DESCRIPTION
**Pull request by fredericbarthelet**
This PR aims to improve `aws-nodejs-typescript` template for the `sls create` command.
It contains the following improvements:
- Added a README.md file explaining how to use the service
- Structured the lambda code base, configuration, mock data and schema into a dedicated sub-folder
- Structured shared lambda code by service in its corresponding sub-folder
- Added typescript config paths for ease of import hereby mentioned resources
- Added middy middleware handler to showcase basic API Gateway REST API workflow on a `POST` request rather than a `GET` request
- Added json-schema-to-ts to benefit from typed input event in lambda code base

### CODE REVIEW

File: lib/plugins/create/templates/aws-nodejs-typescript/README.md
Comment (pgrzesik) [inline lib/plugins/create/templates/aws-nodejs-typescript/README.md:9]: Minor typo: `preffered` -> `preferred`

File: lib/plugins/create/templates/aws-nodejs-typescript/README.md
Comment (pgrzesik) [inline lib/plugins/create/templates/aws-nodejs-typescript/README.md:13]: Minor typo: `dependancies` -> `dependencies`

File: lib/plugins/create/templates/aws-nodejs-typescript/README.md
Comment (pgrzesik) [inline lib/plugins/create/templates/aws-nodejs-typescript/README.md:18]: Minor typo: `dependancies` -> `dependencies`

**Review by pgrzesik — COMMENTED** [review]
Thank you @fredericbarthelet, it looks awesome :bow: I've tested it myself and works like a charm. The only comments I had are about minor typos in the docs, there's however one thing that I'm not sure about - should we include editor-specific files in these types of examples (`.vscode/launch.json`)? I'd love to hear @medikoo opinion on this one

**Comment by fredericbarthelet** [comment]
Hi @pgrzesik and thanks for the review :)
I updated this PR fixing the typos you spotted.

This `.vscode` folder was here beforehand. Its current configuration requires the addition of `serverless-offline` plugin.
Debugger configuration is always useful and I see multiple way we could improve this :
- update the VS Code debugger config to work with `sls invoke local` command rather than serverless-offline
- include serverless-offline as dependency and configure it in `serverless.ts`

In order to make it less VS Code specific, rather than removing the configuration, we could include as well debugger configuration for other IDE, such as WebStorm, with the addition of a `.idea` folder.

It all depends on wether we'd like a rich template, that user can remove unused code from for their specific use case, or wether we'd go for a simple agnostic template, and it's up to the developer to implement his feature. I agree with both approach :)

**Review by medikoo — APPROVED** [review]
> In order to make it less VS Code specific, rather than removing the configuration, we could include as well debugger configuration for other IDE, such as WebStorm, with the addition of a .idea folder.

I also believe that project files should be editor-agnostic. Trying to provide config for every potentially used editor doesn't scale, and it's more related to editor environment than actual project (I believe in this approach: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c)

Anyway in TS matter I'd leave final call to @fredericbarthelet as he also maintains [@serverless/typescript](https://github.com/serverless/typescript/ )

**Comment by pgrzesik** [comment]
Hey @fredericbarthelet :wave: 

> It all depends on wether we'd like a rich template, that user can remove unused code from for their specific use case, or wether we'd go for a simple agnostic template, and it's up to the developer to implement his feature. I agree with both approach :)

I don't have a very strong opinion about this to be honest, rich templates can be great, but sometimes they can also be a little bit confusing for newcomers. In this particular case, I think it makes sense to leave `.vscode`, especially that it was there already. I also like your suggestion to potentially improve it so it works with `sls invoke local` instead of relying on an extra plugin. As for other editors, I agree with @medikoo.  What do you think? Let me know if you'd like to update the `.vscode` configuration before merging, otherwise it looks great to me :1st_place_medal:

**Comment by fredericbarthelet** [comment]
@medikoo @pgrzesik thanks for both your feedbacks.
I removed `.vscode` folder and mention of this folder in the README.md documentation. No IDE-specific code base in the template :)

Let's merge this if you're fine with it ;)

**Review by pgrzesik — APPROVED** [review]
Awesome :tada:

**Comment by tjmcewan** [comment]
Heya, thanks for the awesome project template.  Really appreciate the help, especially as a TS newbie.  I have a couple of questions/comments & I apologise if this isn't where these should be.

I noticed the `ForkTsCheckerWebpackPlugin` config was removed from `webpack.config.js`.  Is this because the functionality is replaced by something else or integrated in another way?  It's just that the package is still referenced in `package.json`.  I'm not sure if this package is working as it is.  I added eslint & suddenly everything is broken (missing types), so I can tell it's working then! :)

Also, should we add a `.gitignore` to the template?  Potentially at least
```
.serverless
.webpack
```

Thanks again! <3

**Comment by fredericbarthelet** [comment]
Hi @tjmcewan and thank you for your feedback, it's always nice to see the template being used in real world use cases :) !

Both your comments are very useful :
- regarding `ForkTsCheckerWebpackPlugin`, this PR removes the lines in `webpack.config.js` related to its configuration. But, as you can see, those lines were commented out and simply removed to make it cleaner. You should open a new issue to address/discuss wether or not this plugin should be brought back in the template
- regarding `.gitignore`, I believe you can proceed with opening a PR adding this straight away :)

Thanks

**Pull request merged**
